app/models/katello/host/subscription_facet.rb

```diff
@@ -91,11 +91,20 @@ def self.find_or_create_host(organization, rhsm_params)
       end
 
       def self.propose_name_from_facts(facts)
-        facts['network.fqdn'] || facts['network.hostname-override'] || facts['network.hostname']
+        setting_fact = Setting[:register_hostname_fact]
+        if !setting_fact.blank? & facts[setting_fact]
+          facts[setting_fact]
+        else
+          Rails.logger.warn(_("register_hostname_fact set for %s, but no fact found.")) % setting_fact unless setting_fact.blank?
```

Comment:
In this scenario, the following error is generated to the user (e.g. during sub-man register):
```
undefined method `%' for true:TrueClass
```

To confirm, the setting was set to 'network.invalid' or 'invalid'